core/src/test/scala/org/apache/spark/scheduler/DAGSchedulerSuite.scala

```diff
@@ -3847,11 +3886,248 @@ class DAGSchedulerSuite extends SparkFunSuite with TempLocalSparkContext with Ti
 
     // Job successful ended.
     assert(results === Map(0 -> 11, 1 -> 12))
+  }
+
+  test("SPARK-33701: shuffle adaptive merge finalization") {
```

Comment:
This specific testcase is not testing adaptive merge finalization - but rather, normal merge finalization works even when adaptive merge is enabled.
Can you add that to the test description ?